hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/lifecycle/OMLifecycleConfigurationCreateRequest.java

```diff
@@ -0,0 +1,169 @@
+    CreateLifecycleConfigurationRequest request =
+        getOmRequest().getCreateLifecycleConfigurationRequest();
+    LifecycleConfiguration lifecycleConfiguration =
+        request.getLifecycleConfiguration();
+
+    OmUtils.validateVolumeName(lifecycleConfiguration.getVolume(), ozoneManager.isStrictS3());
+    OmUtils.validateBucketName(lifecycleConfiguration.getBucket(), ozoneManager.isStrictS3());
+
+    CreateLifecycleConfigurationRequest.Builder newCreateRequest =
+        request.toBuilder();
+
+    LifecycleConfiguration.Builder newLifecycleConfiguration =
+        lifecycleConfiguration.toBuilder();
+
+    newLifecycleConfiguration.setCreationTime(Time.now());
+    newCreateRequest.setLifecycleConfiguration(newLifecycleConfiguration);
+
+    return getOmRequest().toBuilder().setUserInfo(getUserInfo())
+        .setCreateLifecycleConfigurationRequest(newCreateRequest.build())
+        .build();
+  }
+
+  @Override
+  public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, ExecutionContext context) {
+    final long transactionLogIndex = context.getIndex();
+    OMMetadataManager metadataManager = ozoneManager.getMetadataManager();
+
+    CreateLifecycleConfigurationRequest createLifecycleConfigurationRequest =
+        getOmRequest().getCreateLifecycleConfigurationRequest();
+
+    LifecycleConfiguration lifecycleConfiguration =
+        createLifecycleConfigurationRequest.getLifecycleConfiguration();
+
+    String volumeName = lifecycleConfiguration.getVolume();
+    String bucketName = lifecycleConfiguration.getBucket();
+
+    OMResponse.Builder omResponse = OmResponseUtil.getOMResponseBuilder(
+        getOmRequest());
+
+    AuditLogger auditLogger = ozoneManager.getAuditLogger();
+    UserInfo userInfo = getOmRequest().getUserInfo();
+
+    String bucketKey = metadataManager.getBucketKey(volumeName, bucketName);
+    IOException exception = null;
+    boolean acquiredBucketLock = false;
+    OMClientResponse omClientResponse = null;
+    Map<String, String> auditMap = new HashMap<>();
+
+    try {
+      if (ozoneManager.getAclsEnabled()) {
+        checkAcls(ozoneManager, OzoneObj.ResourceType.BUCKET, OzoneObj.StoreType.OZONE,
+            IAccessAuthorizer.ACLType.ALL, volumeName, bucketName, null);
+      }
+
+      OmLifecycleConfiguration omLifecycleConfiguration =
+          OmLifecycleConfiguration.getFromProtobuf(lifecycleConfiguration);
+      auditMap = omLifecycleConfiguration.toAuditMap();
+
+      mergeOmLockDetails(metadataManager.getLock()
+          .acquireWriteLock(BUCKET_LOCK, volumeName, bucketName));
+      acquiredBucketLock = getOmLockDetails().isLockAcquired();
+      omLifecycleConfiguration.valid();
+
+      if (!metadataManager.getBucketTable().isExist(bucketKey)) {
+        LOG.debug("bucket: {} in volume: {} doesn't exist", bucketName,
+            volumeName);
+        throw new OMException("Bucket doesn't exist", BUCKET_NOT_FOUND);
```

Comment:
Shall we distinguish here, whether it's bucket not found, or volume not found?

Reply:
The volume may not exist too, but I think report is OK too.